grpc-api/src/main/java/io/deephaven/grpc_api/table/ExportedTableUpdateListener.java

```diff
@@ -188,19 +164,20 @@ private synchronized void sendUpdateMessage(final Ticket ticket, final long size
             return;
         }
 
-        if (nextMessage == null) {
-            nextMessage = ExportedTableUpdateBatchMessage.newBuilder();
-            liveTableMonitor.addNotification(flushNotification);
-        }
-
         final ExportedTableUpdateMessage.Builder update = ExportedTableUpdateMessage.newBuilder()
                 .setExportId(ticket).setSize(size);
 
         if (error != null) {
             update.setUpdateFailureMessage(GrpcUtil.securelyWrapError(log, error).getMessage());
         }
 
-        nextMessage.addUpdates(update);
+        try {
+            responseObserver.onNext(update.build());
+        } catch (final RuntimeException err) {
+            log.error().append(logPrefix).append("failed to notify listener of state change: ").append(err).endl();
+            session.unmanageNonExport(this);
+            destroy();
```

Comment:
I think you are using liveness when you don't want to. Also, your ListenerImpl need not override destroy.